File: cambio-service/src/main/java/com/jj/cambioservice/config/OpenApiConfig.java
package com.jj.cambioservice.config;

import io.swagger.v3.oas.annotations.OpenAPIDefinition;
import io.swagger.v3.oas.annotations.info.Info;
import io.swagger.v3.oas.models.Components;
import io.swagger.v3.oas.models.OpenAPI;
import io.swagger.v3.oas.models.info.License;

@OpenAPIDefinition(
    info = @Info(
      title = "Cambio Service API",
      version = "1.0",
      description = "Documentation of Cambio Service API"
    )
)
public class OpenApiConfig {

    public OpenAPI customOpenAPI() {
      return new OpenAPI()
        .components(new Components())
          .info(
            new io.swagger.v3.oas.models.info.Info()
              .title("Cambio Service API")
              .version("1.0")
              .license(
                new License()
                  .name("JJ Softwares")
                  .url("https://github.com/josimar-junior/microservices-springcloud-springboot-docker/tree/main/cambio-service")
              )
          );
    }
}

File: book-service/src/main/kotlin/com/jj/bookservice/config/OpenApiConfig.kt
package com.jj.bookservice.config

import io.swagger.v3.oas.annotations.OpenAPIDefinition
import io.swagger.v3.oas.annotations.info.Info
import io.swagger.v3.oas.models.Components
import io.swagger.v3.oas.models.OpenAPI
import io.swagger.v3.oas.models.info.License
import org.springframework.context.annotation.Bean

@OpenAPIDefinition(
    info = Info(
        title = "Book Service API",
        version = "1.0",
        description = "Documentation of Book Service API"
    )
)
class OpenApiConfig {

    @Bean
    fun customOpenAPI(): OpenAPI {
        return OpenAPI()
            .components(Components())
            .info(
                io.swagger.v3.oas.models.info.Info()
                    .title("Book Service API")
                    .version("1.0")
                    .license(
                        License()
                            .name("JJ Softwares")
                            .url("https://github.com/josimar-junior/microservices-springcloud-springboot-docker/tree/main/book-service")
                    )
            )
    }
}